scripts/hostmap-crtsh.nse

```diff
@@ -88,31 +61,51 @@ local function is_valid_hostname (name)
   return true
 end
 
+-- TRUE subdomain check (ends with .parent)
+local function is_subdomain(name, parent)
+  return name:match("%." .. parent .. "$") ~= nil
+end
+
 local function query_ctlogs(hostname)
   local url = string.format("https://crt.sh/?q=%%.%s&output=json", hostname)
   local response = http.get_url(url)
+
   if not (response.status == 200 and response.body) then
     stdnse.debug1("Error: Could not GET %s", url)
     return
   end
+
   local jstatus, jresp = json.parse(response.body)
   if not jstatus then
-    stdnse.debug1("Error: Invalid response from %s", url)
+    stdnse.debug1("Error: Invalid JSON from %s", url)
     return
   end
+
+  local strict = stdnse.get_script_args("hostmap-crtsh.strict")
   local hostnames = {}
+
   for _, cert in ipairs(jresp) do
-    local names = cert.name_value;
+    local names = cert.name_value
     if type(names) == "string" then
       for _, name in ipairs(stringaux.strsplit("%s+", names:lower())) do
-        -- if this is a wildcard name, just proceed with the static portion
+
+        -- strip wildcard
```

Comment:
Leave the original comment. Focus your PR on the core enhancement, avoiding incidental changes.